### PR TITLE
run tests in github action

### PR DESCRIPTION
**Pull request by dsschult**
Apparently we weren't running them with the new setup action.

### CODE REVIEW

**Review by ric-evans — CHANGES_REQUESTED** [review]
While you're at it, can we move all the circleci tests over to here?

**Comment by dsschult** [comment]
> While you're at it, can we move all the circleci tests over to here?

I left that as further work for #69.

**Comment by dsschult** [comment]
FYI: circleci uses requirements-dev.txt, while this uses `.[tests]`.  This was also the reason why the tests were passing for RSA, yet prod wasn't working #67.

**Comment by dsschult** [comment]
@ric-evans  Plan is to merge #67, then update and merge this (so the tests pass before merging).  That all works locally.

**Review by ric-evans — APPROVED** [review]

File: tests/unit_client/client_test.py [hunk 1]
```diff
@@ -185,6 +185,11 @@ def handler(signum, frame):  # type: ignore[no-untyped-def]
 def _jsonify(val: bytes) -> Any:
     return json.loads(val.strip()) if val.strip() else None
 
+def _json_stream(iterable: Iterable) -> Generator[Any, None, None]:
```
Comment (ric-evans) [inline tests/unit_client/client_test.py:189]: a `Generator[Any, None, None]` can be written as `Iterator[Any]`

**Pull request merged**